tekdoc/main.go

```diff
@@ -17,15 +17,72 @@ limitations under the License.
 package main
 
 import (
+	"flag"
 	"fmt"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+)
 
-	_ "github.com/tektoncd/pipeline/pkg/pod"
+var (
+	filename = flag.String("f", "", "Name of the file to parse")
 )
+func check(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+type taskStruct struct {
+	Name   string
+	Description string
+	Default *v1alpha1.ArrayOrString
+	Type v1alpha1.ParamType
+	Type1 v1alpha1.PipelineResourceType
+}
 
 func main() {
-	fmt.Println("Hello this is STILL TekDoc")
-
 	// TODO: take a file path, parse it as YAML -> v1alpha1.Task
+	flag.Parse()
+	dat, err := ioutil.ReadFile(*filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var task v1alpha1.Task
+	if _, _, err := scheme.Codecs.UniversalDeserializer().Decode(dat, nil, &task); err != nil {
+		log.Fatal(err)
+	}
+
 	// TODO: use Go templates to print v1alpha1.Task as Markdown
-	// TODO: tests!
+	fmt.Println("#",task.Name)
+	fmt.Printf("## Install the Task \n kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/%s/%s.yaml\n ### Inputs \n ***Parameters :-***\n",task.Name,task.Name)
+
+	for _, i := range task.Spec.Inputs.Params {
+		InputParams := taskStruct{Name: i.Name, Description: i.Description, Default: i.Default, Type: i.Type}
+		tmpl := template.Must(template.New("Details").Parse("- Name : {{.Name}}, {{.Description}}, ({{.Default}})\n" ))
```

Comment:
You could potentially merge all of these into a single template if you wanted to!

I think it's fine to leave as is for now, but something to consider for the next PR! ^_^

Reply:
This is the next step on which I've started working on :D